Fixes #852: Undocumented Things

joshaidan opened the pull request:
I was studying the Liquid code and saw #852. Figured I would be a good open source citizen and submit a quick pull request.

A few things about this PR. I really wasn't sure where things should go. Like should `comments` go on their own page, or a section on another page, like Introduction. Also, I wasn't sure if there should be a section on forloop helper variables added, as there's a note about them on the iterations page. So give me your feedback and I'll fix it up.

Lastly, for the life of me I could not figure out how to get the {% raw %} {% endraw %} tags to show up in the Input preview of the raw.md page. I know there has to be a way to do it, as it's done on [this](https://help.shopify.com/themes/liquid/tags/theme-tags#raw) Shopify documentation. If someone can let me know how to fix that properly, I'll correct the preview.

fw42 commented:
@admhlt, could you review this?

joshaidan commented:
Hey, before this gets merged just want to draw your attention to the raw.md file (as mentioned in the original post):

 ```text
 {% raw %}
 {% raw %}
   In Handlebars, {{ this }} will be HTML-escaped, but {{{ that }}} will not.
 {% endraw % }
 {% endraw %}
 ```

I put a space after the `% }` in the input {% endraw %} tag because I didn't know how to get liquid not to process that tag so that it would show up in the rendered output. I know there has to be a way to do it as it's done in the Shopify documentation.

admhlt commented:
Hi Josh,

The super-hacky way to do that is to use HTML entities for the percent sign `&#37;` and curly braces, `&#123;` so that Liquid doesn't recognize them. I can do that for you in this PR if you like. I'm taking a look at this now. Thanks for helping out!

joshaidan commented:
I was looking for curly bracket html entities, but couldn't find them. Thanks.

Okay, so if I do:
```
&#123;&#37; raw &#37;&#125;
  In Handlebars, {{ this }} will be HTML-escaped, but {{{ that }}} will not.
&#123;&#37; endraw &#37;&#125;
```
It seems to HTML escape the entities. I could do this:

```
<p class="code-label">Input</p>
<pre class="highlight">
<code>
&#123;&#37; raw &#37;&#125;
  In Handlebars, {{ this }} will be HTML-escaped, but {{{ that }}} will not.
&#123;&#37; endraw &#37;&#125;
</code>
</pre>
```
And it works fine.

Also, I noticed the output is wrong on that page, so I'll fix it up.

joshaidan commented:
Okay, I changed it to the use the pre and code tags, and also corrected the output.

admhlt commented on tags/comments.md:
````diff
@@ -0,0 +1,22 @@
+<p class="code-label">Input</p>
+```liquid
+{% raw %}
+Any contents that you put between {% comment %} and {% endcomment %} tags
````
Can we change `Any contents` to `Anything`?

admhlt commented on tags/comments.md:
```diff
@@ -0,0 +1,22 @@
+---
+title: Comments
```
Perhaps the title (and filename) of this page should be singular `Comment` to match the other pages in this section.

admhlt commented on tags/raw.md:
```diff
@@ -0,0 +1,22 @@
+description: An overview of raw tags in the Liquid template language.
+---
+
+Raw temporarily disables tag processing. This is useful for generating content
```
Good job here — getting Liquid to render Liquid tags is one of the trickiest things we have to deal with in this documentation. :)

joshaidan commented:
Okay, I changed the title and filename of Comment. Also changed `Any contents` to `Anything` and removed the `that` because I don't think it's needed there.

admhlt commented:
Looks great! Thank you!
